packages/server-wallet/src/evm-validator.ts

```diff
@@ -27,16 +27,16 @@ export const validateTransitionWithEVM = async (
     });
     return false;
   }
+
   const bytecode =
     bytecodeCache[from.appDefinition] ??
     (bytecodeCache[from.appDefinition] =
       (await AppBytecode.getBytecode(defaultConfig.chainNetworkID, from.appDefinition, tx)) ||
       MISSING);
+
   if (bytecode === MISSING) {
-    // logger.warn(
-    //   `No bytecode found for appDefinition ${from.appDefinition} and chain id ${config.chainNetworkID}. Skipping valid transition check`
-    // );
-    return true;
+    logger.warn(`No bytecode found for appDefinition ${from.appDefinition}.`);
```

Comment:
This means a warning gets logged in every state update. That should not be the behaviour here -- it should get logged at most once.

Reply:
The state update would fail, too. The interface for this function is expecting a `boolean` response though, so we should log the reason why it returned `false` somewhere I think

Reply:
Oh I see. You're suggesting, then, that the _behaviour_ of the validator should change so that it fails when the bytecode is not known?

In that case, I'd probably expect it to throw an error, instead of returning `false`. Returning `false`, to me, should indicate that it is _known_ to be an invalid transition.

Reply:
It seems like either behaviour is ok, but that the package is not ready for this changeset (eg. since the tests are failing!)

Reply:
I'll leave this PR open for now as we might be in a position to make this change soon... and certainly will have to for mainnet :)